### DISCUSSION TITLE
Duplicate AMS device, can't figure out how to remove it

### DISCUSSION THREAD

**Discussion by longrackslabs** [greghesp/ha-bambulab#629] [General]
Recently i noticed I have a duplicate entries for my AMS2 in HA (shows in devices, the Overview dashboard, but oddly *not* in entities). The one that is working normal in HA, and the other seems dead/zombied.  I can't figure out how to remove this duplicate, and I am afraid to touch too much in fear of breaking other things.

![image](https://github.com/user-attachments/assets/a90de439-4240-4828-a42b-e246dbbd43e5)
![image](https://github.com/user-attachments/assets/c4749215-d72b-4144-8c56-5945c5a8d6ea)

**Comment by AdrianGarside** [comment]
Please try v2.0.27-dev2 where I added logic to remove stale AMS entries after you remove an AMS and/or add it to another printer such that the indices bambu assigns change for the existing attached AMS(s). Your screen shots suggest you did the latter since it looks like what I saw when I just did that myself :). I found my fix is not perfect - I had to restart home assistant a second time after switching to the version with the fix as the first time it removed one of the AMS's incorrectly from the printer with 3 entries (2 correct, 1 stale) - but it comes back on restart and then that issue doesn't hit again.
